RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -120,29 +113,110 @@ public async Task GetBrewStatus_WhenCoffeeReady_ShouldReturn200()
+        mockHttp.When("*")
+            .Respond(HttpStatusCode.InternalServerError);
+
+        var httpClient = mockHttp.ToHttpClient();
+        httpClient.BaseAddress = new Uri(MOCKED_HTTPCLIENT_BASE_URL);
+        var response = await GetEndpointResponse(httpClient: httpClient);
+
+        response.StatusCode.ShouldBe(HttpStatusCode.BadGateway);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Theory]
+    [InlineData(30, BREW_READY_HOT_MESSAGE)]
+    [InlineData(30.01, "Your refreshing iced coffee is ready")]
+    public async Task GetBrewStatus_WhenTemperatureRetrieved_ShouldReturnCorrectMessage(float temperature, string expectedMessage)
+    {
+        var httpClient = CreateMockedHttpClient(temperature);
+        var response = await GetEndpointResponse(httpClient: httpClient);
+
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
+        var content = await response.Content.ReadFromJsonAsync<BrewStatusResponse>();
+        content.ShouldNotBeNull();
+        content.Message.ShouldBe(expectedMessage);
+    }
+
+    private Task<HttpResponseMessage> GetEndpointResponse(
+        IDateTimeProvider? dateTimeProvider = null,
+        IBrewStatusProvider? brewStatusProvider = null,
+        IMemoryCache? cache = null,
+        HttpClient? httpClient = null
+    )
     {
         var mockFactory = _factory.WithWebHostBuilder(builder =>
         {
-            builder.ConfigureTestServices(servicesConfiguration);
+            builder.ConfigureTestServices(services =>
+            {
+                if (dateTimeProvider != null)
+                    services.AddSingleton<IDateTimeProvider>(dateTimeProvider);
+                if (brewStatusProvider != null)
+                    services.AddSingleton<IBrewStatusProvider>(brewStatusProvider);
+                if (cache != null)
+                    services.AddSingleton<IMemoryCache>(cache);
+
+                httpClient ??= CreateMockedHttpClient(30);
+                services.AddHttpClient<IWeatherProxy, WeatherProxy>(client => new WeatherProxy(
+                    Mock.Of<ILogger<WeatherProxy>>(),
+                    httpClient
+                ));
+            });
         });
 
         var client = mockFactory.CreateClient();
         return client.GetAsync("brew-coffee");
     }
+
+    private static HttpClient CreateMockedHttpClient(float temperature)
+    {
+        var responseJson = JsonSerializer.Serialize(new TemperatureResponse
+        {
+            Weather = new CurrentWeather
+            {
+                Temperature = temperature
+            }
+        });
+
+        var mockHttp = new MockHttpMessageHandler();
+        mockHttp.When("*")
+            .Respond(MediaTypeNames.Application.Json, responseJson);
+
+        var httpClient = mockHttp.ToHttpClient();
+        httpClient.BaseAddress = new Uri(MOCKED_HTTPCLIENT_BASE_URL);  // URL not actually called, just needs a base address
```

Comment:
- My weapon of choice for mocking HttpClient is [mockhttp](https://github.com/richardszalay/mockhttp).
- Even a "mocked" HttpClient needs a valid base URL to be instantiated, otherwise it throws a run-time validation error
- The URL we pass in (in this case `MOCKED_HTTPCLIENT_BASE_URL`) is never actually called